ext/web/16_image_data.js

```diff
@@ -0,0 +1,174 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+import * as webidl from "ext:deno_webidl/00_webidl.js";
+import DOMException from "ext:deno_web/01_dom_exception.js";
+const primordials = globalThis.__bootstrap.primordials;
+const {
+  TypeError,
+} = primordials;
+
+const PredefinedColorSpace = [
+  "srgb",
+  "display-p3",
+];
+
+class ImageData {
+  /** @type {number} */
+  #width;
+  /** @type {height} */
+  #height;
+  /** @type {Uint8Array} */
+  #data;
+  /** @type {'srgb' | 'display-p3'} */
+  #colorSpace;
+
+  constructor() {
+    webidl.requiredArguments(
+      arguments.length,
+      2,
+      'Failed to construct "ImageData"',
+    );
+
+    const [arg0, arg1, arg2, arg3] = arguments;
```

Comment:
Use indexed destructuring here as array destructuring using iterators:

{ 0: arg0, 1: arg1, 2: arg2, 3: arg3 }

Reply:
this is also inherently wrong: this shouldnt use destructuring of the `arguments` object since that will conflict with the constructor's length. 
This instead should accept the 4 arguments directly in the constructor, with the latter 2 having a default value (as such, not being part of the constructor's length).